src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.java

```diff
@@ -154,32 +139,17 @@ public ProviderEvaluation<Value> getObjectEvaluation(
 
     final FlagPath flagPath = getPath(key);
 
-    final Struct.Builder evaluationContext = Struct.newBuilder();
-    ctx.asMap()
-        .forEach(
-            (mapKey, mapValue) -> {
-              evaluationContext.putFields(mapKey, TypeMapper.from(mapValue));
-            });
-
-    // add targeting key as a regular value to proto struct
-    if (!StringUtil.isNullOrEmpty(ctx.getTargetingKey())) {
-      evaluationContext.putFields(TARGETING_KEY, Values.of(ctx.getTargetingKey()));
-    }
-
+    final Struct evaluationContext = OpenFeatureUtils.convertToProto(ctx);
     // resolve the flag by calling the resolver API
     final ResolveFlagsResponse resolveFlagResponse;
     try {
       final String requestFlagName = "flags/" + flagPath.getFlag();
+
       resolveFlagResponse =
-          stub.withDeadlineAfter(DEADLINE_AFTER_SECONDS, TimeUnit.SECONDS)
-              .resolveFlags(
-                  ResolveFlagsRequest.newBuilder()
-                      .setClientSecret(clientSecret)
-                      .addAllFlags(List.of(requestFlagName))
-                      .setEvaluationContext(evaluationContext.build())
-                      .setSdk(Sdk.newBuilder().setId(SDK_ID).setVersion(SDK_VERSION).build())
-                      .setApply(true)
-                      .build());
+          confidence
+              .withContext(ConfidenceValue.Struct.fromProto(evaluationContext))
```

Comment:
Add the evaluation context under the key "openFeature" and do the "spread" in FlagResolverClient before fetching.

Reply:
```suggestion
              .withContext(Map.of("openFeature", ConfidenceValue.Struct.fromProto(evaluationContext)))
```

Reply:
picking this up

Reply:
Fixed in [02cd529](https://github.com/spotify/confidence-openfeature-provider-java/pull/88/commits/02cd529e38340eab344766f9c8d369db17381749)